babel/src/main/java/org/apache/calcite/sql/babel/SqlCreateFunctionSqlForm.java

```diff
@@ -0,0 +1,172 @@
+import org.apache.calcite.sql.SqlDataTypeSpec;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlSpecialOperator;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import java.util.List;
+
+public class SqlCreateFunctionSqlForm extends SqlCreate {
+  public final SqlIdentifier functionName;
+  public final SqlIdentifier specificFunctionName;
+  public final List<SqlIdentifier> fieldNames;
+  public final List<SqlDataTypeSpec> fieldTypes;
+  public final SqlDataTypeSpec returnsDataType;
+  public final DeterministicType isDeterministic;
+  public final ReactToNullInputType canRunOnNullInput;
+  public final boolean hasSqlSecurityDefiner;
+  public final int typeInt;
+  public final SqlNode returnExpression;
+
+  private static final SqlSpecialOperator OPERATOR =
+      new SqlSpecialOperator("CREATE FUNCTION", SqlKind.CREATE_FUNCTION);
+
+  /** Creates a SqlCreateFunctionSqlForm.
+   * @param pos position
+   * @param replace if "or replace" token occurred
+   * @param functionName the name of the function
+   * @param specificFunctionName an optional specific functionName
+   * @param fieldNames function parameter names
+   * @param fieldTypes function parameter types
+   * @param returnsDataType return type of the function
+   * @param isDeterministic if "deterministic" is specified
+   * @param canRunOnNullInput if "called on null input" or "returns null on null input"
+   *                          is specified
+   * @param hasSqlSecurityDefiner if "sql security definer" is specified
+   * @param typeInt integer value after inline type
+   * @param returnExpression the expression that is returned
+   * */
+  public SqlCreateFunctionSqlForm(final SqlParserPos pos, final boolean replace,
+      final SqlIdentifier functionName,
+      final SqlIdentifier specificFunctionName, final List<SqlIdentifier> fieldNames,
+      final List<SqlDataTypeSpec> fieldTypes,
+      final SqlDataTypeSpec returnsDataType, final DeterministicType isDeterministic,
+      final ReactToNullInputType canRunOnNullInput, final boolean hasSqlSecurityDefiner,
+      final int typeInt, final SqlNode returnExpression) {
+
+    super(OPERATOR, pos, replace, false);
+    this.functionName = functionName;
+    this.fieldNames = fieldNames;
+    this.fieldTypes = fieldTypes;
+    this.specificFunctionName = specificFunctionName;
+    this.returnsDataType = returnsDataType;
+    this.isDeterministic = isDeterministic;
+    this.canRunOnNullInput = canRunOnNullInput;
+    this.hasSqlSecurityDefiner = hasSqlSecurityDefiner;
+    this.typeInt = typeInt;
+    this.returnExpression = returnExpression;
+  }
+
+  @Override public List<SqlNode> getOperandList() {
+    return ImmutableNullableList.of(functionName, specificFunctionName, returnsDataType,
```

Comment:
Why only return a subset of the fields in the operands list?

Reply:
I think the fix would be adding FieldNames and FieldTypes to the OperandList, is there more to be added?

Reply:
I'm not quite sure what the recommendation is regarding getOperandList. It seems to be primarily referenced in SqlCall's clone() method.

At the very least, we should be declaring a getOperator() override if we have getOperandList() declared.

Reply:
getOperator is added https://github.com/googleinterns/calcite/pull/79/commits/a27c1db25901e8d949eb7c29e0ab2057089146d1.

Reply:
After looking at the documentation of [SqlOperator](https://calcite.apache.org/javadocAggregate/org/apache/calcite/sql/SqlOperator.html) and some [other implementation of class implemented SqlCreate](https://github.com/googleinterns/calcite/blob/master/babel/src/main/java/org/apache/calcite/sql/babel/SqlCreateTable.java), I believe it would be reasonable to add FieldNames and FieldTypes to Operand in the format of SqlNodeList.

Reply:
Which means I would take [Andrew's advice](https://github.com/googleinterns/calcite/pull/79#discussion_r441524252) and make FieldNames and FieldTypes into the type SqlNodeList, which can be added directly into Operand list.

Reply:
Fixes made in https://github.com/googleinterns/calcite/pull/79/commits/55b082e08f165ae803e9ebfc6c1bfa2a29e2b37d.